lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -716,10 +739,17 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
     if (params->size()) {
       interleave(*params,
                  [&](const ParamDecl *param) {
-                   print(param->getInterfaceType(), OTK_None, param->getName(),
+                   OptionalTypeKind optionalKind = OTK_None;
+                   Type objTy = param->getInterfaceType();
+                   if (FD->getAttrs()
+                           .hasAttribute<ImplicitlyUnwrappedOptionalAttr>()) {
```

Comment:
This should be `param`, right?

Reply:
Yeah that's clearly bogus, and also manages to not set optionalKind. I suspect we already have a test that would show this, but it requires the changes to actually remove the IUO type from the type system for the problem to show up.

I'll double check that before merging, though.